pandas/core/generic.py

```diff
@@ -6065,19 +6071,43 @@ def asof(self, where, subset=None):
 
         Returns
         -------
-        where is scalar
+        `where` is scalar
 
-          - value or NaN if input is Series
-          - Series if input is DataFrame
+            - value or `NaN` if input is Series
+            - Series if input is DataFrame
 
-        where is Index: same shape object as input
+        `where` is :class:`pandas.Index`:
```

Comment:
Not sure I understand this. Is an Index for `where` treated differently than other inputes?